### DISCUSSION TITLE
Font/Non English Chars questions

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#4149] [Q&A]
My app may need to use non-english characters (any language) based on configuration. I can't tell in advance which languages will be needed.
Since it runs on an embedded device with limited memory, I don't want to load a large font that supports all languages.

- Is there a way at runtime to supply Slint with font data? (not from std file system since I don't have an std file system on my device, only SD-card)?
- I noticed in my app that if I provide data from configuration that includes non-english characters, they are not shown in controls. However, if I hardcode text using those characters, the specific characters I used are available. As if Slint takes specific characters font as needed at compile time. Since my data will come from configuration I need all chars to be available, how can I do that?
- Is there a way for me to know which font was loaded by Slint in both cases above?

**Comment by tronical** [comment]
I've filed https://github.com/slint-ui/slint/issues/4250 on the context of SD-Card support with fonts. We've run across this a few times, so it's time to track this :)

> As if Slint takes specific characters font as needed at compile time.

Yes, that is currently the same - the character set is decided at compile time. A way out of this would be a central configuration file as in https://github.com/slint-ui/slint/issues/267, where one could specify what kind of glyph coverage is required for the application (for example by writing system/script).
